### PR TITLE
Tidy code and use pre-commit

### PR DESCRIPTION
**Pull request by inverse**
- Leverage pre-commit for linting
- Add pre-commit pipeline
- Tidy type references
- Use black formatting

### CODE REVIEW

File: lambda/lambda_function.py [hunk 1]
```diff
@@ -4,17 +4,17 @@
...
-import logging
-import ask_sdk_core.utils as ask_utils
 import json
+import logging
 
-from ask_sdk_core.skill_builder import SkillBuilder
-from ask_sdk_core.dispatch_components import AbstractRequestHandler
-from ask_sdk_core.dispatch_components import AbstractExceptionHandler
+import ask_sdk_core.utils as ask_utils
+from ask_sdk_core.dispatch_components import (
```
Comment (paulotruta) [inline lambda/lambda_function.py:11]: Would it be possible to lint but keep single module per line imports?
This is not critical but in general doing a single module import per line makes merge conflict resolution simpler

File: lambda/lambda_function.py [hunk 2]
```diff
@@ -4,17 +4,17 @@
...
 
-import logging
-import ask_sdk_core.utils as ask_utils
 import json
+import logging
 
-from ask_sdk_core.skill_builder import SkillBuilder
-from ask_sdk_core.dispatch_components import AbstractRequestHandler
-from ask_sdk_core.dispatch_components import AbstractExceptionHandler
+import ask_sdk_core.utils as ask_utils
```
Comment (paulotruta) [inline lambda/lambda_function.py:10]: if we are not importing python system modules or entire pip libraries, I would prefer we always use the from `from x import y as z`

**Review by paulotruta — CHANGES_REQUESTED** [review]
Left a couple comments regarding imports. More specifically I would like to be sure that the linter does not enforce the single import of multiple modules under the same namespace.

Other than that, excellent additions 🙌

File: lambda/lambda_function.py (hunk 2 above)
Comment (inverse) [reply lambda/lambda_function.py]: Fixed :+1: was like that initially so it didnt change :)

File: lambda/lambda_function.py (hunk 1 above)
Comment (inverse) [reply lambda/lambda_function.py:11]: This one isn't so easy - I think it's a limitation of how Black groups code

**Comment by inverse** [comment]
@paulotruta updated and merged your changes in :+1:

File: lambda/lambda_function.py (hunk 1 above)
Comment (paulotruta) [reply lambda/lambda_function.py:11]: Reading on the Black website...

> The following notable open-source projects trust Black with enforcing a consistent code style: pytest, tox, Pyramid, Django, Django Channels, Hypothesis, attrs, SQLAlchemy, Poetry, PyPA applications (Warehouse, Bandersnatch, Pipenv, virtualenv), pandas, Pillow, Twisted, LocalStack, every Datadog Agent Integration, Home Assistant, Zulip, Kedro, OpenOA, FLORIS, ORBIT, WOMBAT, and many more.
> 
> The following organizations use Black: Facebook, Dropbox, KeepTruckin, Mozilla, Quora, Duolingo, QuantumBlack, Tesla, Archer Aviation.

I think this are enough amazing examples for me to drop my initial request 😄 Let's use Black and see how it works for us.
Fyi: I read the [code style here](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#pragmatism), quite good page explaining the reasonings behind all choices. Pragmatism is the main word, and that feels good to me 👍

**Review by paulotruta — APPROVED** [review]
LGTM 🚀

**Pull request merged**